pkg/certificate/reconciler.go

```diff
@@ -0,0 +1,35 @@
+package certificate
+
+import (
+	"context"
+	"time"
+)
+
+// CheckAndUpdate checks for a MRC to reach a desired condition and updates the status when the condition is met
+func (r *MRCReconciler) CheckAndUpdate(ctx context.Context, mrcClient MRCClient, checkInterval time.Duration) {
+	ticker := time.NewTicker(checkInterval)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+				log.Info().Msgf("Stopping status reconciliation loop for MRC %s", r.mrcName)
+				return
+			case <-ticker.C:
+				mrc := mrcClient.Get(r.mrcName)
+				if mrc != nil {
+					log.Error().Msgf("failed to get MRC %s in status reconciliation loop", r.mrcName)
+				}
+
+				if !r.checkStatus(ctx, mrc) {
+					continue
+				}
+
+				if err := r.updateStatus(ctx, mrc); err != nil {
+					log.Warn().Err(err).Msgf("failed to update status of MRC %s in status reconciliation loop. Condition was met", r.mrcName)
```

Comment:
What happens in this case?  Are there any safe guards to ensure the loop gets restarted?  Is it left the implementor of UpdateMRCStatus to ensure proper number of retries are completed?

Reply:
Per our offline conversation, I have added an error return value to `checkStatus`. An error will be returned if the mrc reconciler loop should exit. For example, the loop should terminate if another control plane replica executing the same loop, sees the condition is met and updates the status. The other loops waiting for the same condition may never see the condition in the desired state, but they will see the status as having already been updated, signifying the loop can terminate. A false bool and nil error will be returned if the condition has not been met but the check should be retried.